### ISSUE TITLE
actions-on-google types not working fix

### ISSUE THREAD

**Issue by Iridio** [dialogflow/dialogflow-fulfillment-nodejs#33]
If I use your dependency of actions-on-google (2.0.0-alpha4) when using SimpleResponse I get the following error:
`TypeError: dialogflow_fulfillment_1.SimpleResponse is not a constructor
`
If instead I change your dependecy to version 2.0.1 everything works just fine.

My code is as follow:

```
import { WebhookClient, SimpleResponse } from 'dialogflow-fulfillment'

let result = //My code to get a result string
            if (app.requestSource === app.ACTIONS_ON_GOOGLE) {
                let conv = app.conv()
                conv.close(new SimpleResponse({
                    speech: `<speak>
                                           <prosody rate="100%" pitch="-2st">
                                            My name is <prosody rate="slow">Wonder Woman</prosody>.
                                            </prosody>
                                        </speak>`, text: result
                }))
                app.add(conv)
            }
            else
                app.add(result)
```
Would you like a PR or is enough this issue?

**Comment by paracycle** [comment]
@matthewayne I just got bitten by the same version problem. If the version of `actions-on-google` SDK used by the host app is not the same as the one used by `dialogflow-fulfillment` then there will be two instances of the `actions-on-google` SDK. The host app will be using its own copy for response types (like `Suggestions`, `SimpleResponse`, etc) and they won't be handled properly in the `Conversation` class of the `actions-on-google` SDK copy of this library, since that code will be trying to match classes from a different module and will fail.

In summary, the current way this library handles `app.conv()` is problematic if the user doesn't take care to depend on the exact same version of `actions-on-google`. A more healthy approach might be to allow the user to pass in the `DialogflowConversation` type into the `conv()` method. That would allow the library to not have to depend on the `actions-on-google` library (this is the only dependency) and would make this functionality work properly.

**Comment by paracycle** [comment]
BTW, here is quick a workaround if anybody else is running into this problem:

Instead of doing this:
```js
const conv = agent.conv();
conv.ask('Test string');
agent.add(conv);
```

do this:
```js
const { DialogflowConversation } = require('actions-on-google');
const conv = new DialogflowConversation(request);
conv.ask('Test string');
agent.add(conv);
```

**Comment by omfd** [comment]
@paracycle how is the above solution going to work. 

agent.conv() returns the same thing as new DialogflowConversation(request);

And how do create these elements(BasicCard,
  BrowseCarousel,
  BrowseCarouselItem,
  Button,
  Carousel,
  Image,
  LinkOutSuggestion,
  List,
  MediaObject,
  Suggestions) from actions library ??

Also I am facing one more issue -- 
**MalformedResponse
'final_response' must be set.**

even **conv.close(new Suggestions('Lets Go !!'))** is not solving the issue instead of _conv.close(new Suggestions('Lets Go !!'))_

**Comment by omfd** [comment]
@paracycle 

`const { DialogflowConversation } = require('actions-on-google');
const conv = new DialogflowConversation(request);
conv.ask('Test string');
**agent.add(conv);**`

This statement throws exception. Understood what you were trying to do(compensate for version difference) in the above statement. 

`

> unknown response type

`

**Comment by omfd** [comment]
```
if (**response instanceof DialogflowConversation**) {
      this.client.addActionsOnGoogle_(response.serialize());
    } else if (response instanceof Suggestion && this.existingSuggestion_(response.platform)) {
      this.existingSuggestion_(response.platform).addReply_(response.replies[0]);
    } else if (response instanceof Payload && this.existingPayload_(response.platform)) {
      throw new Error(`Payload response for ${response.platform} already defined.`);
    } else if (response instanceof RichResponse) {
      this.responseMessages_.push(response);
    } else {
      throw new Error('unknown response type');
    }
```

Because of this statement in the dialogflow-fulfillment package.

**Comment by paracycle** [comment]
@omfd Apologies, you are correct. Since I had ended up using the same version of the AoG library as this library, I couldn't detect the problem with the type check that happens during `agent.add(conv)`.

So ultimately, the only solution right now is to use the exact same version of AoG as this library and to call [`npm dedupe`](https://docs.npmjs.com/cli/dedupe) in your repo to ensure that there is a single version of the AoG library in your dependencies.
